metricflow/plan_conversion/instance_converters.py

```diff
@@ -728,6 +785,7 @@ def transform(self, instance_set: InstanceSet) -> InstanceSet:  # noqa: D102
             dimension_instances=instance_set.dimension_instances,
             time_dimension_instances=instance_set.time_dimension_instances,
             entity_instances=instance_set.entity_instances,
+            group_by_metric_instances=instance_set.group_by_metric_instances,
```

Comment:
Ok, so this works, but only because the lone callsite for removing metrics happens to be hit just above the point where group by metric instances are created.

I think we can file this under "let's clean up the weird transform API usage for simple property updates" effort and come back to it later.